### PR TITLE
[RDY] Sailboats

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "Adds wind-powered sailboats"```

#### Purpose of change
Adding Sailboats to the game!

#### Describe the solution
This is WIP because this is a rough first-pass.
There have been discussions on the discord on how best to implement this, so I'm putting this here as a place to discuss ( probably better ) ways to do it.
My current method is to directly alter the "engine" power depending on the dot product of the wind vector vs facing vector, * by magnitude of wind.
It seems to allow basic tacking, you gain speed by going with the wind, you can turn into the wind and carry some momentum, so with some S-shapes you can kinda go against the wind ( slowly ) 
The utility of sailboats is limited, for one, theres no currently no way , yet, for the player to see wind direction and strength - that will hopefully come with the UI PR's.
Bridges still make pure-boats kinda useless.
Rivers are quite narrow to adequately pull off tacking manoeuvres.
Sailboat utility may expand in the future with the bridge fix, coastlines, oceans , who knows!
Theoretically, landsails should work, although it'd have to be a very light vehicle.

#### Describe alternatives you've considered
mlangsdorf said the best way to approach this would be to directly leverage air resistance in the slowdown() vehicle function, I tried that, but nothing I could finagle worked to my satisfaction.
Instead directly modifying engine power seemed to get a passable result.
I'm open to feedback and balancing and input, infact I probably need it as the current way is probably not fit for purpose yet, but here as a proof of concept.

There was also discussion of more actions involving the sails to more directly represent real sailing.
Trimming, turning and loads of other sailing terminology I don't really know about, 
but this I feel has the same level of representation as a vehicle/bicycle does without gearshifting etc.
There is certainly space for the customization of messages displayed to player, and sailing-specific actions however.

#### Additional context
N/A

### CODE REVIEW

File: data/mods/Boats/b_recipes.json [hunk 1]
```diff
@@ -24,6 +24,20 @@
...
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_VEHICLES",
+    "skill_used": "fabrication",
+    "difficulty": 2,
+    "time": 30000,
+    "reversible": true,
+    "autolearn": true,
+    "qualities": [ { "id": "HAMMER", "level": 2 }, { "id": "SAW_W", "level": 1 } ],
+    "using": [ [ "sewing_standard", 100 ] ],
+    "components": [ [ [ "rag", 40 ], [ "sheet", 2 ] ], [ [ "2x4", 10 ], [ "stick", 5 ] ], [ [ "nail", 30 ] ] ]
```
Comment (mlangsdorf) [inline data/mods/Boats/b_recipes.json]: This seems like an excellent opportunity to use a long stick or two.

**Comment by davidpwbrown** [comment]
This will need to be updated after #28143 is merged

**Comment by davidpwbrown** [comment]
Updated to new wind override system, rebased onto master, technically complete, although I am not sure if it fits the criteria for what we want for sailboats.

**Comment by kevingranade** [comment]
I  might just be dumb, but I can't sort out how to install this, I spawned a raft and it wouldn't let me install it anywhere but on the seat, and even then only once I removed the paddle.

**Comment by kevingranade** [comment]
I have a raft with paddles swapped out for a sail, and it's complaining about not enough wheels while in the water.

**Pull request merged**